src/app/_hooks/useSetParams.js

```diff
@@ -23,5 +27,5 @@ export function useSetParam(param) {
     }
     router.replace(`?${newParams.toString()}`);
   };
-  return { add, replace, remove };
+  return { add, replace, set: replace, remove };
```

Comment:
could you elaborate on these changes? Why do you need two methods which do the same?

Reply:
`replace` to keep existing usages ( if any ) and `set` i like more as it shorter, and i see it often in similar api's
so its just adding alias to existing function

Reply:
What do you think about some refactorings - 
```
export function useSetParam(param) {
  const searchParams = useSearchParams();
  const router = useRouter();
  const newParams = new URLSearchParams(searchParams);
  // TODO: to avoid writing template literals several times - big chance to make a mistake
  const replaceWithNewParams = () => router.replace(`?${newParams.toString()}`);

  return {
    // TODO: if function uses only in one place - it's better to inline it
    add: value => {
      newParams.append(param, value);
      replaceWithNewParams();
    },

    set: value => {
      newParams.delete(param);
      // TODO: to avoid indentation
      const values = Array.isArray(value) ? value : [value];
      values.forEach(v => newParams.append(param, v));
      replaceWithNewParams();
    },

    remove: value => {
      // TODO: Please cant it always be - newParams.delete(param, value);
      // OR at least - newParams.delete(param, value || undefined);
      if (value) {
        newParams.delete(param, value);
      } else {
        newParams.delete(param);
      }
      replaceWithNewParams();
    },
  };
}
```